### DISCUSSION TITLE
Quasar Cli enable all Quasar components in quasar.conf.js for runtime compiler

### DISCUSSION THREAD

**Discussion by xlaussel** [quasarframework/quasar#11532] [General - Components / Directives / etc]
Hello,

I am using Quasar in CLI mode and I am compiling templates on the fly into the browser: I have the vueCompiler option set to true into the build section of the quasar.conf.js in order to enable the runtime compiler.

It works fine but the Quasar components used in the on the fly compiled templates are unknown to the compiler by default. I have to register them one by one into the components option of the framework section of the quasar.conf.js file.

I would like to register them all.

I did some googling and tried some undocumented options (importStrategy: 'auto', import: 'all') but it doesn't work.
The doc says that filling “components” and “directives” is required only if “all” is set to false but this 'all' option is not documented. I tried to put all: true  but it doesn't work.
The comments into the framework section refers to "the auto-import strategy" but I don't tell how to activate it.

Before the Qusar CLI I was using my own webpack conf. I was activating the Quasar plugin by calling app.use(Quasar) and after this call all the quasar components were available to the runtime compiler.

Any idea of how to proceed?

**Comment by hawkeye64** [comment]
I believe it is `importStrategy: 'all'`. When it is set to auto, then the components are found in your code and used to make an import array.

**Reply by xlaussel** [reply]
> I believe it is `importStrategy: 'all'`. When it is set to auto, then the components are found in your code and used to make an import array.

Thank you but I tried (in the framework section) and It doesn't work :-(

**Reply by hawkeye64** [reply]
I see this in the upgrade guide:

```
Remove quasar.conf.js > framework > importStrategy. Auto import works so great that is now used by default and as the only option.
```

You should raise a ticket on this issue. I don't see any other way for your code to work.
